### PR TITLE
[tests] use `torch_device` instead of `auto` for model testing 

### PR DESCRIPTION
**Pull request by faaany**
## What does this PR do?
When running test cases under the models folder on XPU, I found that many model tests fail at the same test `test_model_parallel_beam_search`, e.g. 
```bash
FAILED tests/models/bigbird_pegasus/test_modeling_bigbird_pegasus.py::BigBirdPegasusModelTest::test_model_parallel_beam_search - RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cpu and xpu:0! (when che
```
This is because `device_map="auto"` is used. As elaborated in [this](https://github.com/huggingface/transformers/pull/29309) PR, the device_map="auto" mechanism is still not mature on XPU, causing model to be loaded on CPU, rather than on XPU.

If there is no particular reason for using `auto`, I would suggest using `torch_device` instead `auto`, because `torch_device` is more specific than `auto` and we don't have the need to use `auto`, e.g. large model inference, in our tests anyway. WDYT? @ArthurZucker

### CODE REVIEW

**Comment by amyeroberts** [comment]
Hi @faaany - thanks for opening this PR! Using device_map="auto" is necessary for this test - it's checking that beam search works when the model is split across devices. If it doesn't work with XPU, then you can add a skip with unittest e.g.

```py
if "xpu" in torch_device:
    return unittest.skip("device_map='auto' does not work with XPU devices")
```

**Comment by faaany** [comment]
Hi @amyeroberts , thanks so much for reviewing this PR! I used `torch_device=="xpu"` instead of `'xpu' in torch_device`, hope this is fine for you. And yes, I will remove this skip once `auto` works on XPU.

File: tests/generation/test_utils.py [hunk 1]
```diff
@@ -1073,6 +1073,9 @@ def test_beam_search_generate_dict_outputs_use_cache(self):
     @require_torch_multi_accelerator
     def test_model_parallel_beam_search(self):
         for model_class in self.all_generative_model_classes:
+            if torch_device == "xpu":
```
Comment (amyeroberts) [inline tests/generation/test_utils.py:1076]: is better, as the device can be numbered

```suggestion
            if "xpu" in torch_device:
```

**Review by amyeroberts — COMMENTED** [review]
Thanks for handling!

Just a small nit

**Comment by faaany** [comment]
> Thanks for handling!
> 
> Just a small nit

done, thx!

**Review by amyeroberts — APPROVED** [review]
Thanks!

**Pull request merged**